core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/store/dot-favorite-page.store.ts

```diff
@@ -0,0 +1,138 @@
+                const individualPermissions = {
+                    READ: []
+                };
+                individualPermissions.READ = formData.permissions
+                    ? formData.permissions.map((role: DotRole) => role.id)
+                    : [];
+                individualPermissions.READ.push(formData.currentUserRoleId, CMS_OWNER_ROLE_ID);
+
+                return this.dotTempFileUploadService.upload(file).pipe(
+                    switchMap(([{ id, image }]: DotCMSTempFile[]) => {
+                        if (!image) {
+                            return throwError(
+                                this.dotMessageService.get(
+                                    'favoritePage.dialog.error.tmpFile.upload'
+                                )
+                            );
+                        }
+
+                        return this.dotWorkflowActionsFireService.publishContentletAndWaitForIndex<DotCMSContentlet>(
+                            'Screenshot',
+                            {
+                                screenshot: id,
+                                title: formData.title,
+                                url: formData.url,
+                                order: formData.order
+                            },
+                            individualPermissions
+                        );
+                    }),
+                    take(1),
+                    tapResponse(
+                        () => {
+                            this.patchState({ closeDialog: true, loading: false });
+                        },
+                        (error: HttpErrorResponse) => {
+                            this.dotHttpErrorManagerService.handle(error);
+                            return of(null);
+                        }
+                    )
+                );
+            })
+        );
+    });
+
+    constructor(
+        private dotCurrentUser: DotCurrentUserService,
+        private dotRolesService: DotRolesService,
+        private dotMessageService: DotMessageService,
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private dotTempFileUploadService: DotTempFileUploadService,
+        private dotWorkflowActionsFireService: DotWorkflowActionsFireService
+    ) {
+        super(null);
+    }
+
+    /**
+     * Sets initial state data from props, roles and current logged user data
+     *
+     * @param {DotFavoritePageProps} props
+     * @memberof DotFavoritePageStore
+     */
+    setInitialStateData(props: DotFavoritePageProps): void {
+        const propsData = {
+            isAdmin: props.pageState.user.admin,
+            imgWidth: parseInt(props.pageState.params.viewAs.device?.cssWidth, 10) || 1024,
+            imgHeight:
+                parseInt(props.pageState.params.viewAs.device?.cssHeight, 10) ||
+                (parseInt(props.pageState.params.viewAs.device?.cssWidth, 10) || 1024) /
+                    IMG_RATIO_43
+        };
```

Comment:
Still think that is an overhead send the `pageState` to use 3 props.

Reply:
changed!